tests/test_modeltest.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+from pytestqt.qt_compat import QStandardItemModel, QStandardItem, \
+    QFileSystemModel, QStringListModel, QSortFilterProxyModel, QT_API
+
+pytestmark = pytest.mark.usefixtures('qtbot')
+
+
+def test_standard_item_model(qtmodeltester):
+    """
+    Basic test which uses qtmodeltester with a QStandardItemModel.
+    """
+    model = QStandardItemModel()
+    items = [QStandardItem(str(i)) for i in range(5)]
+    items[0].setChild(0, items[4])
+    model.setItem(0, 0, items[0])
+    model.setItem(0, 1, items[1])
+    model.setItem(1, 0, items[2])
+    model.setItem(1, 1, items[3])
+
+    qtmodeltester.check(model)
+
+
+def test_file_system_model(qtmodeltester, tmpdir):
+    tmpdir.ensure('directory', dir=1)
+    tmpdir.ensure('file1.txt')
+    tmpdir.ensure('file2.py')
+    model = QFileSystemModel()
+    model.setRootPath(str(tmpdir))
+    qtmodeltester.check(model)
+    tmpdir.ensure('file3.py')
+    qtmodeltester.check(model)
+
+
+@pytest.mark.skipif(QT_API == 'pyside', reason='For some reason this fails in '
+                                               'PySide with a message about'
+                                               'columnCount being private')
```

Comment:
That's probably one of the reasons why the cast is needed, but there really doesn't seem any way to do it with PySide...

It seems `columnCount` is private in `QAbstractListModel` because it doesn't make any sense (it'll always be 1).

What about adding a `_column_count` method which does something like this (untested)?

```python
def _column_count(self, parent=QModelIndex()):
    if isinstance(self._orig_model, QAbstractListModel):
        assert parent is QModelIndex()
        return 1
    else:
        return self._model.columnCount(parent)
```

and then using that instead of `columnCount` everywhere?